dask/array/core.py

```diff
@@ -2922,8 +2974,6 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
     if len(set(seq_dtypes)) > 1:
         dt = reduce(np.promote_types, seq_dtypes)
         seq = [x.astype(dt) for x in seq]
```

Comment:
Leaving a note here for reference, we should be able to use `_meta` to perform this type promotion thus dropping this now outdated code. Handling this in PR ( https://github.com/dask/dask/pull/4925 ).

Reply:
Agreed, I totally missed that. For the record, there's still quite some work to be done in that file, as well as a few others, as tracked on #4888. Thanks a lot for fixing this @jakirkham! :)